language/stdlib/modules/LibraAccount.move

```diff
@@ -971,12 +965,44 @@ module LibraAccount {
         aborts_if LibraTimestamp::is_genesis()
             && !exists<AccountOperationsCapability>(CoreAddresses::LIBRA_ROOT_ADDRESS())
             with Errors::NOT_PUBLISHED;
-        aborts_if Vector::length(auth_key_prefix) + Vector::length(LCS::serialize(addr)) != 32
-            with Errors::INVALID_ARGUMENT;
+        include CreateAuthenticationKeyAbortsIf;
         // We do not need to specify aborts_if if account already exists, because make_account will
         // abort because of a published FreezingBit, first.
     }
 
+    /// Construct an authentication key, aborting if the prefix is not valid.
+    fun create_authentication_key(account: &signer, auth_key_prefix: vector<u8>): vector<u8> {
+        let authentication_key = auth_key_prefix;
+        Vector::append(
+            &mut authentication_key, LCS::to_bytes(Signer::borrow_address(account))
+        );
+        assert(
+            Vector::length(&authentication_key) == 32,
+            Errors::invalid_argument(EMALFORMED_AUTHENTICATION_KEY)
+        );
+        authentication_key
+    }
+    spec fun create_authentication_key {
+        /// The specification of this function is abstracted to avoid the complexity of
+        /// vector concatenation of serialization results. The actual value of the key
+        /// is assumed to be irrelevant for callers. Instead the uninterpreted function
+        /// `spec_abstract_create_authentication_key` is used to represent the key value.
+        /// The aborts behavior is, however, preserved: the caller must provide a
+        /// key prefix of a specific length. This length is currently arbitrarily
+        /// chosen; this might be refined once LCS has a specification, and if this would
+        /// ever become relevant for verification of callers.
+        pragma opaque;
+        include [abstract] CreateAuthenticationKeyAbortsIf;
+        ensures [abstract]
+            result == spec_abstract_create_authentication_key(auth_key_prefix) &&
+            len(result) == 32;
+    }
+    spec schema CreateAuthenticationKeyAbortsIf {
+        auth_key_prefix: vector<u8>;
+        aborts_if 8 + len(auth_key_prefix) != 32 with Errors::INVALID_ARGUMENT;
```

Comment:
@tzakian : Can you comment on this aborts_if specification?  This is written assuming that the byte representation of the address of an account is 8 bytes.  But, in our conversation just now you indicated that an account is 16 bytes.

Reply:
To add to this, there is no LCS spec where this would be written down AFAICT. I followed the link in LCS.move but it just points to the code.

Reply:
> his is written assuming that the byte representation of the address of an account is 8 bytes. 

More precise, it is assuming that the actual length does not matter for verification for callers, as is written now in the doc comment (and as has been assumed already implicitly before this PR).

Reply:
I'm not fully sure about the semantics of `abstract` so can't fully comment on this. 

To answer Shaz's question, the address is `16` bytes. I am fairly certain the LCS of an account address will be 16 bytes as well since the underlying address is just a `Vec<u8>`, so the 8 looks odd here, but I leave it up to you to decide if this change is needed though since it might change the abort behavior.

Reply:
Ok, I updated this to 16 (before this PR, I think the way how we defined LCS it was actually 4. ;-)).

Notice that there is still the issue I mentioned above: there is no easily discoverable spec of the "canonical" LCS format. At least, I did not find anything then the code when following the link you find in `LCS.move`. This might be OK for this first release, but we need to have this at some point. @matbd @sblackshear